### DISCUSSION TITLE
&Hole and closed doors/windows

### DISCUSSION THREAD

**Discussion by Heino0404** [firemodels/fds#13270] [Q&A]
Hey, I’m new to FDS and still learning how to use it. I have a question about how to model closed doors and windows in FDS. 
Closed Doors: How should I model closed doors in FDS? I understand that the &HOLE command defines an opening, but if I want to represent a closed door, should I use &HOLE or is there a different approach?
Closed Windows: Similarly, how should I handle closed windows? These materials have different properties compared to my walls. Do I need to specify them in the simulation, and if so, how?

My idea is to create a &MATL_ID with the properties of a door and add it to a SURF_ID, and then assign it to the &HOLE, like this:
&MATL ID='door_wood', DENSITY=500, CONDUCTIVITY=0.12, SPECIFIC_HEAT=2.0 /
&SURF ID='door_Surf', COLOR='BROWN', MATL_ID='door_wood', THICKNESS=0.25 /
&Hole ID='closed door', XB=1.0,1.5,1.0,1.25,0.0,2.0, SURF_ID='door_surf' /

Will this work? And similarly for windows, using properties for glass?

I hope you understand my question, and I would appreciate any clarification.

**Comment by drjfloyd** [comment]
As a new user, I suggest making simple inputs to test things out.  Like in this case a mesh that is maybe 20^3 and just put in a wall with a door and window and see if you can open and close them and successfully assign surface properties. 

Also look at the User's Guide and Verification Guide. Between the two there are simple examples for most common things people do with FDS.

**Comment by obscureed** [comment]
Hopefully you've found answers already by experimentation, as drjfloyd suggested. A few comments, though:
• A &HOLE replaces &OBST with fluid -- typically, air. This is why there is no SURF_ID parameter in a &HOLE command.
• See the User Guide for "overlapping obstructions". You can specify the OBST of the wall and the OBST of the door, in that order, and it will work.
• I almost never bother defining different material properties for doors, windows etc. Note that the different conductivity and other properties will not conduct heat from one side of the door to the other (at least, not by default, not unless you have HT3D=T). These material properties are *surface* properties. Heat that leaves one side of a door goes into some void or other, and is lost. The other side of the door also exchanges heat with a different void (or maybe the same one, it's hard to tell).
• With that in mind, a permanently closed door is effectively the same as a wall in my simulations. Doors are only interesting when they open and/or close. Here's an example that does both (applied to a model using 0.1m-resolution mesh, with some semi-paranoid adjustments to the coordinates):
```
&DEVC  XYZ= 1.10,0.80,1., QUANTITY='TIME',ID='devc_time'  /
!!
&HOLE XB=   2.680,   2.820,   5.300,   6.200,   0.000,   2.000, CTRL_ID='ctrl_door_stair_gap_',  /
&OBST XB=   2.680,   2.820,   5.300,   6.200,   0.000,   2.000, COLOR='TAN',   /
&DEVC XB=   2.720,   2.720,   5.300,   6.200,   0.000,   2.000, ID='volflow1_door_stair_gap_',  QUANTITY='U-VELOCITY', SPATIAL_STATISTIC='AREA INTEGRAL',  /
&CTRL ID='ctrl_door_stair_gap_',  FUNCTION_TYPE='CUSTOM', INPUT_ID='devc_time',  RAMP_ID='ramp_door_stair_gap_',  /
&RAMP ID='ramp_door_stair_gap_',  T =    0.000 F = -1 /
&RAMP ID='ramp_door_stair_gap_',  T =  219.900 F = -1 /
&RAMP ID='ramp_door_stair_gap_',  T =  220.000 F = 1 /
!!
&OBST XB=   1.790,   2.710,   6.190,   6.310,  -0.010,   2.010, CTRL_ID='ctrl_door_stair_leaf',  SURF_ID='WALL',  COLOR='TAN' /
&CTRL ID='ctrl_door_stair_leaf',  FUNCTION_TYPE='CUSTOM', INPUT_ID='devc_time',  RAMP_ID='ramp_door_stair_leaf',  /
&RAMP ID='ramp_door_stair_leaf',  T =    0.000 F = -1 /
&RAMP ID='ramp_door_stair_leaf',  T =  219.900 F = -1 /
&RAMP ID='ramp_door_stair_leaf',  T =  220.000 F = 1 /
!!
```

**Reply by drjfloyd** [reply]
@obscureed 

"• I almost never bother defining different material properties for doors, windows etc. Note that the different conductivity and other properties will not conduct heat from one side of the door to the other (at least, not by default, not unless you have HT3D=T). These material properties are surface properties. Heat that leaves one side of a door goes into some void or other, and is lost. The other side of the door also exchanges heat with a different void (or maybe the same one, it's hard to tell)."

This is not the case in the current release. At one point in time the default back face boundary condition was VOID. It has been EXPOSED since FDS 6.1 (2014).  The default behavior In FDS 6.9.1 is that heat is conducted in 1D if the obst is one grid thick or thinner.

&HEAD CHID='ht'/

&MESH XB=0,1,0,1,0,1,IJK=10,10,10/

&TIME T_END=100/

&MATL ID='MY MAT',DENSITY=1000,CONDUCTIVITY=100,SPECIFIC_HEAT=1/

&SURF ID='MY SURF',MATL_ID='MY MAT',THICKNESS=0.002,COLOR='BLACK'/

&SURF ID='HOT',TMP_FRONT=1000,COLOR='RED'/

&OBST XB=0.4,0.5,0.0,1.0,0.0,1.0,SURF_ID='MY SURF'/

&VENT PBX=0,SURF_ID='HOT'/
&VENT PBX=1,SURF_ID='OPEN'/
&VENT PBY=0,SURF_ID='OPEN'/
&VENT PBY=1,SURF_ID='OPEN'/
&VENT PBZ=0,SURF_ID='OPEN'/
&VENT PBZ=1,SURF_ID='OPEN'/

&BNDF QUANTITY='WALL TEMPERATURE'/
![ht_0629](https://github.com/user-attachments/assets/2316940b-b97c-45d5-8bea-51cd74cb8b7b)

**Reply by obscureed** [reply]
@drjfloyd -- Thanks for the correction! (Maybe I should have added a disclaimer that my comments are not official advice.) I had not noticed the important distinction between a 1-cell-thick door and a thicker modelled &OBST. I think I will probably continue to use the same material properties for doors and walls, because they are both fairly good insulators for the timescales and temperatures I deal with, but I should check in more detail.
Windows could be a more difficult question: they are still fairly good insulators (assuming that they are at least double-glazed), but they might or might not allow thermal radiation through. I really do not want that level of complication, and I think it is not necessary in my models. Again, something to think about case by case.

**Reply by zxhm9** [reply]
> "@drjfloyd -- Thanks for the correction! (Maybe I should have added a disclaimer that my comments are not official advice.) I had not noticed the important distinction between a 1-cell-thick door and a thicker modelled &OBST. I think I will probably continue to use the same material properties for doors and walls, because they are both fairly good insulators for the timescales and temperatures I deal with, but I should check in more detail. Windows could be a more difficult question: they are still fairly good insulators (assuming that they are at least double-glazed), but they might or might not allow thermal radiation through. I really do not want that level of complication, and I think it is not necessary in my models. Again, something to think about case by case."



@obscureed 
I am facing the same struggle whilst trying to model a window on a roof that opens at 500°C to simulate ventilation. I want it to be a solid, conductive glass pane when closed and an open vent when triggered.
What I tried:
I defined a glass material and a 'Window' SURF.
I placed this surface on a VENT and used a DEVC/CTRL to open it (F=1.0) at 500°C and close it (F=-1.0).
To make the solid window part of the roof, I placed this vent inside a HOLE in my roof obstruction.
The problem:
The window doesn't seem to close properly. It behaves like it's always open, and flow isn't blocked when the control function says it should be closed. could you pleasw help me with the correct way to model an operable window that is part of a solid surface?

code:

&RAMP ID='Control01_RAMP', T=499.75, F=-1.0/
&RAMP ID='Control01_RAMP', T=500.25, F=1.0/
&CTRL ID='Control01', FUNCTION_TYPE='CUSTOM', RAMP_ID='Control01_RAMP', LATCH=.FALSE., INPUT_ID='Device1364'/
&DEVC ID='Device1364', QUANTITY='TEMPERATURE', XYZ=22.6,2.8,1.8/

&MATL ID='Glass',
      SPECIFIC_HEAT=84.0,
      CONDUCTIVITY=76.0,
      DENSITY=2700.0/

&SURF ID='Window',
      RGB=102,255,102,
      MATL_ID(1,1)='Glass',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=2.0/

&VENT ID='Vent12', SURF_ID='Window', XB=19.875,21.9375,10.46729,11.962617,5.6,5.6, IOR=3, TEXTURE_ORIGIN=19.875,10.46729,5.6, CTRL_ID='Control01'/ 


&HOLE ID='Hole', XB=19.875,21.9375,11.588785,11.775701,5.196226,6.158491/ 
&HOLE ID='Hole', XB=19.875,21.9375,10.46729,10.654206,5.388679,5.966038/ 
&HOLE ID='Hole', XB=19.875,21.9375,11.401869,11.588785,5.388679,6.158491/ 
&HOLE ID='Hole', XB=19.875,21.9375,10.841121,11.401869,5.388679,6.350943/ 
&HOLE ID='Hole', XB=19.875,21.9375,10.654206,10.841121,5.388679,6.543396/ 
&HOLE ID='Hole', XB=19.875,21.9375,11.775701,11.962617,5.773585,6.158491/

**Comment by Heino0404** [comment]
Hi @obscureed and @drjfloyd !

First of all, I’d like to thank both of you for your responses and for the exchange of ideas – I really appreciate the discussion.
Unfortunately, I’m not able to start with simpler cases because I’m working on this as part of my final thesis project for university, so I have to tackle an entire project right away.

Following your advice, I’ve programmed the doors and windows as closed because they won’t have a significant impact on my simulation. For doors that do influence the ventilation, I have used the VENT command.

I also wanted to ask if you could help me with another point or add to this topic. I’ve applied 1D heat transfer calculations throughout, which I understand is the default in FDS. However, for the roof structure, I need to implement 3D heat conduction.
My question is: Can I input objects with different materials when using 1D heat transfer, or is there something specific I need to consider for that?

Apologies if I’m not expressing myself clearly—English is not my first language, but I’m doing my best. If anything is unclear, please feel free to ask for clarification.

Thanks again for your help!

**Reply by mcgratta** [reply]
3-D heat conduction should be used carefully. It is limited. Much depends on whether the obstructions are 0 cells thick or not. You can use multiple material components, but not multiple layers. The only way to use multiple layers is to assign material components to individual obstructions that make up the solid. See the User's Guide and Verification Guide for examples, and try a simple example of your own before embarking on the full simulaiton.

**Reply by obscureed** [reply]
I have been simulating things professionally for some decades, and I still wonder whether it is quicker to go straight to the full model, or whether to build test cases. If you are pressed for time, the temptation is to go straight to the full model. However, if there are features that do not behave as you expect, troubleshooting is massively quicker on small models. It also saves time to read the manual and look for good examples.

**Reply by obscureed** [reply]
For the roof, you say that you need 3D heat conduction, but this is an assumption worth challenging. It sounds like you have some time pressure, so I would advise that you should try as hard as you can to avoid "complex geometry" and 3D conduction and all beta features. Therefore, your roof is likely to be stair-stepped (in other words, an alternating pattern of horizontal OBST then vertical OBST, all grid-aligned). Therefore, 1D heat conduction will cover heat transfer that is perpendicular to each OBST. Are the other directions of heat transfer really critical? It makes a big difference to model complexity. As @mcgratta suggested, if 3D conduction is not fundamental to the question, you should consider simple, good-enough models. It is often not a problem in a university project to admit that you have made big approximations/simplifications. Maybe you should explain to us why you think you need 3D conduction -- your English has been fine so far.

**Reply by Heino0404** [reply]
Yes, I am under time pressure with this project. My Room ist octagonal in shape and spans two floors but there is an open atrium in the center, creating a large open hall with a gallery. I have already simplified the geometry to a rectangular shape.

So far, my approach has been to read through the User Guide while simultaneously building my model. I've also discovered that the &GEOM command wasn’t very useful for my case, so I've approximated everything as best I could using OBST commands. The roof structure of the building consists of open wooden beams. As mentioned, I have constructed the roof in a stepped pattern ro account for the slope.

The main objective of the simulation is to determine the temperature distribution in the roof and at the exposed wooden beams during the fire. Since these beams protrude into the room, I was told that I would need to use 3D heat conduction for this part of the model, as the heat flows around the components.

For the rest of the building, I’m using 1D heat conduction and programmed it like the example below:

&MATL ID='Brick', DENSITY=1800, CONDUCTIVITY=0.8, SPECIFIC_HEAT=0.84 /
&MATL ID='Insulation', DENSITY=100, CONDUCTIVITY=0.04, SPECIFIC_HEAT=1.4 /
&MATL ID='Concrete', DENSITY=2400, CONDUCTIVITY=1.7, SPECIFIC_HEAT=0.84 /
&MATL ID='Wood Paneling', DENSITY=500, CONDUCTIVITY=0.12, SPECIFIC_HEAT=2.0 /
&MATL ID='Air_Layer', DENSITY=1.2, CONDUCTIVITY=0.025, SPECIFIC_HEAT=1.0 /
&MATL ID='PVC', DENSITY=1500, CONDUCTIVITY=0.16, SPECIFIC_HEAT=1.2 /
&MATL ID='Cement_Screed', DENSITY=2000, CONDUCTIVITY=1.4, SPECIFIC_HEAT=0.84 /

&SURF ID='Floor_Slab_Surf', COLOR='SILVER', BACKING='EXPOSED',
MATL_ID(1,1)='PVC',
MATL_ID(2,1)='Cement_Screed',
MATL_ID(3,1)='Insulation',
MATL_ID(4,1)='Concrete',
THICKNESS(1)=0.02,
THICKNESS(2)=0.06,
THICKNESS(3)=0.06,
THICKNESS(4)=0.12 /

Will that work?

And what you mean with "3-D heat conduction ist limited" ?

**Reply by obscureed** [reply]
Simplification of an octagonal shape to rectangular is a good idea. Also, avoiding `&GEOM` makes sense in your case, particularly while it is a beta feature.

An alternative to doing 3D conduction in the beams and the roof would be to accept 1D conduction at the exposed solid surfaces, and to analyse the gas temperatures adjacent to (or near to) these surfaces. This is a much, much simpler model to set up and to run.

What are the approximations involved in this 1D approach?
• Conduction inside the beams is approximated. It follows that the heat transfer rate between the beams and the gas is approximated.
— However, once the focus is on the gas temperatures, the heat transfer inside the beams is a secondary effect. The gas temperatures have a big effect on the beam temperatures, but not the other way around.
• Conduction across the width and height of each beam is not done correctly, and possibly not done at all.
— Assuming the beams are surrounded by similar (not necessarily identical) gas temperatures on all sides, this is not a big issue.
— I have not completely worked out how `BACKING='EXPOSED'` would work here. This would be a question where building a test case will save you time in the end.
— As a first attempt, I might be even more approximate: set the 1D surface material of the exposed beam surfaces to (1/3)*(beam thickness), and apply `BACKING='VOID'`. This would be a good-enough approximation in the middle of the beams' faces, and possibly a poor approximation at the corners. It is easy to set up, and should run without many complications.
• Conduction along the length of the beam is not done at all.
— This could be critical in some situations. Suppose the beams were steel, and there was a localised fire beneath their centres, and the question is about the temperature of the steel where it meets or enters the walls. Then, lengthwise conduction would be critical. Here, maybe it is not.

If you keep thinking about the approximations involved, you might be able to conclude that 1D conduction is all you need in a final model. In fact, the 1D-conduction model is so much simpler that I strongly recommend that you do this first, even if your ambition is to move on to a 3D-conduction model eventually. Your first goal should be to get some results, even horribly approximate results, from a simplistic model. Then you can add in more complexity if you have time. Working on the simpler models will give you experience that is useful in defining the later models. 

One real effect to bear in mind is that when wooden beams are exposed to hot gas, then the outer wood turns to char, which is a better insulator than wood. Usually, the charred wood stays attached to the inner wood for a significant time (assuming that the gas is not hot enough to start combustion in the relevant timescales). Wooden beams therefore often survive fires much better than might be expected. This is a significant effect in reality, but not one that you should think about including in your models.

Good luck!
